dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -260,9 +260,16 @@ BufferedFrame& FramesDecoderGpu::FindEmptySlot() {
   DALI_FAIL("Could not find empty slot in the frame buffer");
 }
 
-
 void FramesDecoderGpu::Reset() {
   SendLastPacket(true);
   FramesDecoder::Reset();
 }
+
+FramesDecoderGpu::~FramesDecoderGpu() {
+  cuvidDestroyVideoParser(nvdecode_state_->parser);
+  cuvidDestroyDecoder(nvdecode_state_->decoder);
```

Comment:
Is there a chance that `nvdecode_state_` or `nvdecode_state_->parser` `nvdecode_state->decoder` are in an invalid state at this point?

Reply:
Are purposedly ignoring errors here? Should we do:
```
  if (nvdecode_state_->parser) {
    NVCUVID_CALL(cuvidDestroyVideoParser(nvdecode_state_->parser));
  }

  if (nvdecode_state_->decoder) {
    NVCUVID_CALL(cuvidDestroyDecoder(nvdecode_state_->decoder));
  }
```
?

Reply:
There isn't any interesting error that we can get here at this point, I think. I didn't want to handle throwing exceptions from this destructor, so I just did it like that.
Basically we do best effort to destroy the parser and decoder and it should work.

Reply:
I wonder if out static analysis won't scream about not checking the return value.